finetune_adapter.py

```diff
@@ -34,24 +35,16 @@
 ds_config = {
     "train_micro_batch_size_per_gpu": micro_batch_size,
     "gradient_accumulation_steps": gradient_accumulation_steps,
-    "zero_optimization": {
-        "stage": 2,
-        # "overlap_comm": True,
-        # "allgather_partitions":True,
-        # "allgather_bucket_size": 2e8,
-        # "reduce_scatter":True,
-        # "contiguous_gradients": True,
-        # "round_robin_gradients": True,
-        "offload_optimizer": {"device": "cpu", "pin_memory": True},
-        "offload_param": {"device": "cpu"},
-    },
+    "zero_optimization": {"stage": 2},
 }
 
 
 def main(
-    data_dir: Path = Path("data/alpaca"), checkpoint_dir: Path = Path("checkpoints/stabilityai/stablelm-base-alpha-3b")
+    data_dir: Path = Path("data/alpaca"),
+    pretrained_dir: Path = Path("checkpoints/stabilityai/stablelm-base-alpha-3b"),
```

Comment:
Can we keep the `checkpoint_dir` name?

Reply:
In Lit-LLaMA, it is also pretrained_path and this is to disambiguate with the checkpoint directory where we save training checkpoints.

Reply:
It is easier to see what this argument expectes if it matches the name in `convert_hf_checkpoint`, `generate`, and other scripts.

Particularly for users who are not knowledgeable about the differences between pertaining, training, fine-tuning etc

Reply:
I reverted back to the ambiguous name but I disagree on every level.

Reply:
I think we should be consistent with lit-llama actually, so I'd personally go with `pretrained_dir` (for the same reason why we chose `pretrained_dir` there)
let's just write a meaningful docstring for that argument so it's clear it coincides with the checkpoint